core/templates/dev/head/domain/skill/skill-mastery-backend-api.service.ts

```diff
@@ -16,56 +16,66 @@
-    var _fetchSkillMasteryDegrees = function(
-        skillIds, successCallback, errorCallback) {
-      $http.get(SKILL_MASTERY_DATA_URL_TEMPLATE, {
-        params: {
-          comma_separated_skill_ids: skillIds.join(',')
-        }
-      }).then(
-        function(response) {
-          if (successCallback) {
-            successCallback(response.data.degrees_of_mastery);
-          }
-        }, function(errorResponse) {
-          if (errorCallback) {
-            errorCallback(errorResponse.data);
-          }
-        });
-    };
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
 
-    var _updateSkillMasteryDegrees = function(
-        masteryPerSkillMapping, successCallback, errorCallback) {
-      var putData = {
-        mastery_change_per_skill: masteryPerSkillMapping
-      };
-      $http.put(SKILL_MASTERY_DATA_URL_TEMPLATE, putData).then(
-        function(response) {
-          if (successCallback) {
-            successCallback();
-          }
-        }, function(errorResponse) {
-          if (errorCallback) {
-            errorCallback(errorResponse.data);
-          }
-        });
-    };
+import { SkillDomainConstants } from 'domain/skill/skill-domain.constants';
 
-    return {
-      fetchSkillMasteryDegrees: function(skillIds) {
-        return $q(function(resolve, reject) {
-          _fetchSkillMasteryDegrees(skillIds, resolve, reject);
-        });
-      },
-      updateSkillMasteryDegrees: function(masteryPerSkillMapping) {
-        return $q(function(resolve, reject) {
-          _updateSkillMasteryDegrees(masteryPerSkillMapping, resolve, reject);
-        });
+@Injectable({
+  providedIn: 'root'
+})
+export class SkillMasteryBackendApiService {
+  constructor(private httpClient: HttpClient) {}
+
+  _fetchSkillMasteryDegrees(skillIds: Array<string>, successCallback: Function,
+      errorCallback: Function): void {
+    this.httpClient.get(SkillDomainConstants.SKILL_MASTERY_DATA_URL_TEMPLATE, {
+      params: {
+        comma_separated_skill_ids: skillIds.join(',')
+      }
+    }).toPromise().then((response: any) => {
+      if (successCallback) {
+        successCallback(response.degrees_of_mastery);
       }
+    }, (errorResponse) =>{
+      if (errorCallback) {
+        errorCallback(errorResponse);
+      }
+    });
+  }
+
+  _updateSkillMasteryDegrees(masteryPerSkillMapping: {[key: string]: number},
+      successCallback: Function, errorCallback: Function): void {
+    let putData = {
+      mastery_change_per_skill: masteryPerSkillMapping
     };
+    this.httpClient.put(SkillDomainConstants.SKILL_MASTERY_DATA_URL_TEMPLATE,
+      putData).toPromise().then((response: any) => {
+      if (successCallback) {
+        successCallback();
+      }
+    }, (errorResponse) => {
+      if (errorCallback) {
+        errorCallback(errorResponse);
+      }
+    });
+  }
```

Comment:
Hi @Showtim3, is this code working fine? I'm asking because I think  `put` and `post` don't work just like that. This is because in this file [https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/App.ts#L112-L139](https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/App.ts#L112-L139) interceptors are added for $http put and post requests. I experienced this issue in my PR #8426 also. I had to [fix it](https://github.com/oppia/oppia/pull/8426/commits/1235eaac1de924817e591f96068aed58980a4c07) creating parameters to match the request made by $http interceptors. Thanks!

Reply:
Yeah, you're right @nishantwrp -- thank you for being active :). 

Please take a look at #8507. Thanks!

Reply:
Ok the #8507 is in.